File: js/49.group-anagrams.js
/*
 * @lc app=leetcode id=49 lang=javascript
 *
 * [49] Group Anagrams
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
  let map = {}
  for (let i=0; i<strs.length; i++) {
    let str = strs[i]
    let strSort = [...str].sort().join('')
    map[strSort] = map[strSort] ? map[strSort] : []
    map[strSort].push(str)
  }

  return Object.values(map)
};
// @lc code=end

File: 49.group-anagrams.py
#
# @lc app=leetcode id=49 lang=python3
#
# [49] Group Anagrams
#

# @lc code=start
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        strsSorted = {}
        for str in strs:
            srt = ''.join(sorted(str))
            if srt not in strsSorted:
                strsSorted[srt] = []
            strsSorted[srt].append(str)
        return strsSorted.values()
# @lc code=end

